### PR TITLE
use Scrutinizer for find and test before clear

### PR DESCRIPTION
**Pull request by ReenExe**
Also test changes http://sandbox.onlinephpfunctions.com/code/b91261fd29da86357e717e7343b37091d1ba3a61

### CODE REVIEW

File: lib/Elastica/Query/Fuzzy.php [hunk 1]
```diff
@@ -35,10 +35,10 @@ public function __construct($fieldName = null, $value = null)
      */
     public function setField($fieldName, $value)
     {
-        if (!is_string($value) or !is_string($fieldName)) {
+        if (!is_string($value) || !is_string($fieldName)) {
             throw new InvalidException('The field and value arguments must be of type string.');
         }
-        if (count($this->getParams()) > 0 and array_shift(array_keys($this->getParams())) != $fieldName) {
+        if ($this->getParams() && key($this->getParams()) != $fieldName) {
```
Comment (merk) [inline lib/Elastica/Query/Fuzzy.php]: Should we be doing a strict comparison here?

Comment (ReenExe) [inline lib/Elastica/Query/Fuzzy.php]: @merk, I stay logic `as is` before - only refactoring, and test it before in `sandbox`

Comment (merk) [inline lib/Elastica/Query/Fuzzy.php]: Whenever code is touched for changes like this without tests it is worth the effort to validate that there is no functionality changed by adding a unit test. Why cant your sandbox code be adapted to a simple unit test?

File: lib/Elastica/Query/Fuzzy.php [hunk 2]
```diff
@@ -57,13 +57,13 @@ public function setFieldOption($param, $value)
     {
         //Retrieve the single existing field for alteration.
         $params = $this->getParams();
-        if (count($params) < 1) {
+        if (empty($params)) {
```
Comment (ruflin) [inline lib/Elastica/Query/Fuzzy.php]: I prefer to keep the count(..) here for the "meaning" of the code. Behaviour should be the same. Perhaps even more meaningful would be `count(...) == 0`

**Comment by ruflin** [comment]
@ReenExe Would be nice if you could add a unit test. Otherwise perhaps remove the more critical change that was discussed.

**Comment by ReenExe** [comment]
@ruflin, Ok, see please https://github.com/ruflin/Elastica/pull/1142

**Pull request closed without merging**